main.go

```diff
@@ -209,6 +217,10 @@ func main() {
 			return
 		}
 
+		if cfg.auth.Authentication.Header.Enabled {
+			req.Header.Set(cfg.auth.Authentication.Header.XForwardedFor, forwardedFor)
```

Comment:
This looks a bit dangerous. By a mistake you can override some existing headers. Is that intentional?